k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -182,6 +184,24 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
             AsyncStream.fromFuture(f)
         }
       }
+      // Ignore any cases where we receive a resource version lower
+      // than the last received resource version, by creating a sliding
+      // two-item window over the stream of watch events.
+      stream.paired
```

Comment:
I don't think this works.  Consider:

stream = [3, 1, 2]
paired = [(x,3), (3,1), (1,2)]
filtered = [(x,3), (1,2)]
mapped = [3,2]

I think you could probably accomplish this with a scanLeft...

```
stream.scanLeft[(W, Boolean)]((Zero, false)) { case ((largest, incr), w) =>
  if (w > largest) (w, true)
  else (largest, false)
}.filter(_._2).map(_._1)
```

or something like that

Reply:
Yeah, you're right; 0208ec44f22688d16fd14786916f25736b78daf4 should be more correct.